File: heavy.cpp
#include <iostream>
#include <iomanip>
#include <cmath>

double run(int size){
  double sum = 0;
  for( int y=0 ; y<size ; ++y ){
    for( int x=0 ; x<size ; ++x ){
      sum += std::sin(x+y+sum);
      sum += std::cos(x-y+sum);
    }
  }
  return sum;
}

int main(){
  int size;
  std::cin >> size;
  std::cout << std::setprecision(20) << run(size) << std::endl;
  return 0;
}

File: go/heavy.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
	"strconv"
)

func run(size int) float64 {
	sum := 0.0
	for y := 0; y < size; y++ {
		for x := 0; x < size; x++ {
			sum += math.Sin(float64(x+y) + sum)
			sum += math.Cos(float64(x-y) + sum)
		}
	}
	return sum
}

func main() {
	stdin := bufio.NewScanner(os.Stdin)
	stdin.Scan()
	text := stdin.Text()
	size, _ := strconv.ParseInt(text, 10, 32)
	fmt.Printf("%.20f\n", run(int(size)))
}
